community_supplied/Linecard/netconf-fpc-cpu-stats.rule

```diff
@@ -0,0 +1,261 @@
+/*
+ * Detects FPC CPU and memory utilization and notifies when anomalies are found.
+ * One input control detection
+ * 
+ *   1) "threshold" is the threshold that causes the rule to report
+ *      an anomaly.  By default it's 80. This rule will set a dashboard
+ *      color to red when *all* the CPU/Buffer memory/ Heap memory utilization greater than
+ *      'threshold'. 
+ *   2) slot-name,  is a regular expression that matches the
+ *      FPC slot number that you would like to monitor.  By default it's
+ *      all slots [0-9]*. 
+ */
+healthbot {
+    topic linecard.fpc  {
+        rule check-netconf-fpc-cpu-memory-stats {
+	    synopsis "FPC CPU, Memory Heap & Buffer Utilization change detector";
+            description "Collects CPU, Memory Heap& Buffer Utilization details from FPC slot and notify anomalies based on threshold values ";
+	        /*
+            * Use FPC slot as key for rule.
+            */
+            keys slot;
+            sensor fpc-sensor {
+			    synopsis "FPC CPU Memory sensor initialization";
+                description "FPC iAgent sensor to collect data from network device";
+                iAgent {
+                    file fpc-cpu-memory-stats.yml;
+                    table FPCCpuMemoryStatsTable;
+                    frequency 300s;              
+                }
+            }
+            /*
+            * Fields defined with variables, dynamic thresholds and constant values
+            */
+            field dt-buffer-util {
+                formula {
+                    dynamic-threshold {
+                        algorithm k-means;
+                        learning-period 7d;
+                        pattern-periodicity 1d;
+                        field-name "$memory-buffer-utilization";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for memory buffer utilization";
+            }
+            field dt-cpu-total {
+                formula {
+                    dynamic-threshold {
+                        algorithm k-means;
+                        learning-period 7d;
+                        pattern-periodicity 1d;
+                        field-name "$cpu-total";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for CPU utilization";
+            }
+            field dt-heap-util {

```

Comment:
CAn we use descriptive names now that we don't have the artificial limitations? How about dt-heap-utilization?

Reply:
Modified